src/DynamoUtilities/ObservableHashSet.cs

```diff
@@ -24,8 +23,16 @@ public void Add(T item)
 
         public void RemoveWhere(Predicate<T> match)
         {
-            set.RemoveWhere(match);
-            CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove, null));
+            List<T> removedObjs = new List<T>();
+            foreach(var item in set.ToArray())
+            {
+                if (match(item))
+                {
+                    removedObjs.Add(item);
+                    set.Remove(item);
+                }
+            }
+            CollectionChanged?.Invoke(this, new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Remove, removedObjs));
```

Comment:
Looks like passing null to that specific `NotifyCollectionChangedEventArgs` constructor throws an exception.
@aparajit-pratap I tried to pass in the removed objects here. PTAL

Reply:
Yes, I put in a fix here: https://github.com/DynamoDS/Dynamo/pull/12719

Reply:
I think this is overkill at least now when we're not using any of this information.